src/dra/75F54.c

```diff
@@ -178,7 +178,74 @@ INCLUDE_ASM("asm/us/dra/nonmatchings/75F54", func_80116B0C);
 
 INCLUDE_ASM("asm/us/dra/nonmatchings/75F54", func_801177A0);
 
-INCLUDE_ASM("asm/us/dra/nonmatchings/75F54", func_80117AC0);
+void func_80117AC0(void) {
+    Collider sp10;
+    s32 collisionCount;
+    CheckCollision(PLAYER.posX.i.hi, PLAYER.posY.i.hi + 0x19, &sp10, 0);
+    collisionCount = (s32)sp10.unk0 & 1;
+    CheckCollision(PLAYER.posX.i.hi + 4, PLAYER.posY.i.hi + 0x19, &sp10, 0);
+    if ((s32)sp10.unk0 & 1) {
+        collisionCount += 1;
+    }
+    CheckCollision(PLAYER.posX.i.hi - 4, PLAYER.posY.i.hi + 0x19, &sp10, 0);
+    if ((s32)sp10.unk0 & 1) {
+        collisionCount += 1;
+    }
+    if ((g_Player.pl_vram_flag & 0x41) == 0x41) {
+        collisionCount += 1;
+    }
+    PLAYER.unk1E = 0;
+    func_8010E27C();
+    if (collisionCount == 0) {
+        func_8010E7AC();
+        return;
+    }
+    switch (PLAYER.step_s) {
+    case 0:
+        if (PLAYER.animFrameDuration < 0) {
+            func_8010DA48(0xF6);
+            PLAYER.step_s += 1;
+        }
+        return;
+    case 1:
+        // If any directional button is pressed
+        if (g_Player.padTapped & 0xF000) {
+            func_8010DA48(0xC8);
+            PlaySfx(0x6EE);
+            PLAYER.step_s = 0;
+            return;
+        }
+        if (g_Player.unk72 == 1) {
+            PLAYER.animFrameIdx = 0;
+            PLAYER.animFrameDuration = 3;
+            return;
+        }
+        if (g_Player.unk72 == 2) {
+            if (PLAYER.animFrameIdx != 0) {
+                PLAYER.animFrameIdx = 1;
+                PLAYER.animFrameDuration = 3;
+            }
+            return;
+        }
+        if (g_Player.unk72 == 3) {
+            if (PLAYER.animFrameIdx >= 2) {
+                PLAYER.animFrameIdx = 2;
+                PLAYER.animFrameDuration = 3;
+                PLAYER.step = 2;
+                PLAYER.step_s = 5;
+            }
+            return;
+        }
+        if (g_Player.unk72 == 4) {
```

Comment:
```suggestion
        } else if (g_Player.unk72 == 4) {

Reply:
Is this a code style standard for the repo or personal preference? I've seen both styles (if-return if-return versus if-else if) and would be nice to know what code suggestions are suggestions versus requirements

Reply:
I'm fine with either one but this style seems to be what @Xeeynamo recommended you in his version of the scratch:
https://decomp.me/scratch/e6syb

Reply:
Particularly here https://github.com/Xeeynamo/sotn-decomp/pull/243/files/7998f1c2cc3bcb61a45c117c62f1e4b048f117c5#r1220486401 an early-return was preferred rather than the nesting

Reply:
I think that style is only in his version of the scratch because that's what I had in my old version of the scratch, I changed it to this way in my local copy :)

Reply:
regarding the early return, if it saves a lot of code enclosure i think they are preferable, regarding parenthesis, i think it's just a matter of preference, i think the parenthesis provide a clear separation between relational and logical operators, and it makes it clear for me to read them that way, that's how i see they write in every decomp project i've put my eyes on.